compose/src/commonMain/kotlin/ElementView.kt

```diff
@@ -73,15 +83,55 @@ public fun <T> ElementView(
                 }
             }
         }
+
+        // Dirty hack for correctly handling hover after taps. Android doesn't always propagate the
+        // release event properly, instead sending a "move" event even after the release happens,
+        // and while Compose's internal tap recognition works around that you can't reuse the
+        // implementation if you need to receive move events while waiting for it. However, we can
+        // still leverage their workaround by setting a flag on tap and treating our event as a
+        // different one. This is only necessary for touch events - the desktop user with a mouse
+        // should maintain hover state after releasing a tap.
+        var treatNextTouchAsExit by remember { mutableStateOf(false) }
         Kanvas(
             Modifier
                 .matchParentSize()
-                .pointerInput(null) {
+                .pointerInput(null) { // hover events must be processed manually
```

Comment:
Could this while loop and the detectTapGestures functionality be combined into the same `.pointerInput` call?

Reply:
Nope. `detectTapGestures` does the same sort of loop internally, so whichever one comes first blocks the other one from running.